processing/src/main/java/io/druid/query/dimension/PushDownQueryDimensionSpec.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.query.dimension;
+
+
+import io.druid.query.extraction.ExtractionFn;
+import io.druid.segment.DimensionSelector;
+import io.druid.segment.column.ValueType;
+
+public class PushDownQueryDimensionSpec implements DimensionSpec
```

Comment:
The dimension name for this dimension spec is the the output name of the delegate. This dimension spec is only used in case of nested query push down.
`
 @Override
 public String getDimension()
{    // the dimension name is same as the output name.
    return delegate.getOutputName();
}
`